forch/faucet_event_client.py

```diff
@@ -105,23 +105,37 @@ def has_event(self, blocking=False):
             else:
                 return False
 
-    def _filter_faucet_event(self, event):
-        event_id = int(event.get('event_id'))
-        if event_id <= self._last_event_id:
-            LOGGER.debug('Outdated faucet event #%d', event_id)
+    def _filter_faucet_event(self, event, target_event):
+        if not self._check_event_order(event):
+            return False
+        if not target_event:
             return False
-        self._last_event_id += 1
-        if event_id != self._last_event_id:
-            raise Exception('Out-of-sequence event id #%d' % event_id)
+        return True
+
+    def _check_event_order(self, event):
+        event_id = int(event.get('event_id'))
+        if not event.get('debounced'):
```

Comment:
if event.get('debounced'):
  return True

Less indentation is better.  Early exit is better.

event_id check can be moved after debounce check, and should remove .get (because then event_id should always be defined, right?).